server/appointment/reschedule/clientReschedule.php

```diff
@@ -0,0 +1,267 @@
+	$response = array();
+	$response['success'] = true;
+
+	try {
+		$clientInformation = validateClientInformation($token, $firstName, $lastName, $emailAddress, $phoneNumber);
+		
+		$appointmentId = $clientInformation['appointmentId'];
+		$appointmentAccessor = new AppointmentAccessor();
+		$appointmentAccessor->rescheduleAppointment($appointmentId, $appointmentTimeId);
+
+		$noteAccessor = new NoteAccessor();
+		$notes = $noteAccessor->addNote($appointmentId, 'Rescheduled by client [Automatic Note]');
+
+		$response['message'] = AppointmentConfirmationUtilities::generateAppointmentConfirmation($appointmentId);
+	} catch (Exception $e) {
+		$response['success'] = false;
+		$response['error'] = $e->getCode() === MY_EXCEPTION ? $e->getMessage() : 'There was an error on the server rescheduling your appointment. Please refresh the page and try again.';
+	}
+
+	echo json_encode($response);
+}
+
+function cancelAppointmentWithToken($token, $firstName, $lastName, $emailAddress, $phoneNumber) {
+	$response = array();
+	$response['success'] = true;
+
+	try {
+		$clientInformation = validateClientInformation($token, $firstName, $lastName, $emailAddress, $phoneNumber);
+		
+		$appointmentId = $clientInformation['appointmentId'];
+		$appointmentAccessor = new AppointmentAccessor();
+		$appointmentAccessor->cancelAppointment($appointmentId);
+
+		$noteAccessor = new NoteAccessor();
+		$notes = $noteAccessor->addNote($appointmentId, 'Cancelled by client [Automatic Note]');
+	} catch (Exception $e) {
+		$response['success'] = false;
+		$response['error'] = $e->getCode() === MY_EXCEPTION ? $e->getMessage() : 'There was an error on the server cancelling the appointment. Please try again.';
+	}
+
+	echo json_encode($response);
+}
+
+function emailConfirmationWithToken($token, $firstName, $lastName, $emailAddress, $phoneNumber) {
+	$response = array();
+	$response['success'] = true;
+
+	try {
+		$clientInformation = validateClientInformation($token, $firstName, $lastName, $emailAddress, $phoneNumber);
+
+		$appointmentId = $clientInformation['appointmentId'];
+		$confirmationMessage = AppointmentConfirmationUtilities::generateAppointmentConfirmation($appointmentId);
+		
+		if (PROD && isset($clientInformation['email'])) {
+			EmailUtilities::sendHtmlFormattedEmail($clientInformation['email'], 'VITA Appointment Rescheduled Confirmation', $confirmationMessage, 'noreply@vita.unl.edu');
+		} else {
+			$response['message'] = $confirmationMessage;
+		}
+	} catch (Exception $e) {
+		$response['success'] = false;
+		$response['error'] = $e->getCode() === MY_EXCEPTION ? $e->getMessage() : 'There was an error on the server sending the email. Please try again.';
+	}
+
+	echo json_encode($response);
+}
+
+/* 
+ * Private functions
+ */
+
+function isAppointmentValidForReschedule($appointmentInformation) {
+	// See if it has been cancelled
+	$isAppointmentCancelled = isset($appointmentInformation['cancelled']) && $appointmentInformation['cancelled'] == true;
```

Comment:
Just wanna throw this out because it is something that I continually run into and forget about shortly thereafter... But MySQL actually returns 0/1 for false/true respectively, and so we can't use `===` to compare the result field value and a boolean literal because then we're comparing an integer to a boolean which is false in the `===` world. So one can explicitly cast to the same data type and use `===` or alternatively use `==`, but I think I prefer explicitly casting and then using `===`.